athena-vertica/athena-vertica.yaml

```diff
@@ -75,37 +72,50 @@ Resources:
           disable_spill_encryption: !Ref DisableSpillEncryption
           spill_bucket: !Ref SpillBucket
           spill_prefix: !Ref SpillPrefix
-          data_bucket: !Ref DataBucket
           export_bucket: !Ref ExportBucket
           default_vertica: !Ref VerticaConnectionString
 
       FunctionName: !Sub "${AthenaCatalogName}"
       Handler: "com.amazonaws.connectors.athena.vertica.VerticaCompositeHandler"
       CodeUri: "./target/athena-vertica-1.0.jar"
-      Description: "A guided example for writing and deploying your own federated Amazon Athena connector for a custom source."
+      Description: "Amazon Athena Vertica Connector"
       Runtime: java8
       Timeout: !Ref LambdaTimeout
       MemorySize: !Ref LambdaMemory
       Policies:
         - Statement:
             - Action:
                 - athena:GetQueryExecution
-                - s3:ListAllMyBuckets
-                - secretsmanager:GetSecretValue
-                - ec2:CreateNetworkInterfaces
-                - sts:GetSessionToken
-                - iam:CreateUser
               Effect: Allow
               Resource: '*'
           Version: '2012-10-17'
-          #S3CrudPolicy allows our connector to spill large responses to S3. You can optionally replace this pre-made policy
-          #with one that is more restrictive and can only 'put' but not read,delete, or overwrite files.
-        - S3CrudPolicy:
+        - Statement:
+            - Action:
+                - s3:ListBucket
+              Effect: Allow
+              Resource:
+                - !Sub 'arn:aws:s3:::${ExportBucket}'
+                - !Sub 'arn:aws:s3:::${SpillBucket}'
+          Version: '2012-10-17'
+        - Statement:
+            - Action:
+                - secretsmanager:GetSecretValue
+              Effect: Allow
+              Resource: !Sub 'arn:aws:secretsmanager:*:*:secret:${SecretNameOrPrefix}'
+        - S3ReadPolicy:
             BucketName:
               Ref: SpillBucket
-          #S3CrudPolicy allows our connector to spill large responses to S3. You can optionally replace this pre-made policy
-          #with one that is more restrictive and can only 'put' but not read,delete, or overwrite files.
-        - S3CrudPolicy:
+        - S3WritePolicy:
+            BucketName:
+              Ref: SpillBucket
+        - S3ReadPolicy:
+            BucketName:
+              Ref: ExportBucket
+        - S3WritePolicy:
             BucketName:
               Ref: ExportBucket
-        #VPCAccessPolicy allows our connector to run in a VPC so that it can access your data source.
+      #VPCAccessPolicy allows our connector to run in a VPC so that it can access your data source.
+      VpcConfig:
+        SecurityGroupIds:
+          - !Ref LambdaSecurityGroup
```

Comment:
Why is this needed? can we add some comments?

Reply:
SecurityGroup that should be applied to the Connector Lambda function to enable it to connect to the VPC of Vertica database

Reply:
Ah, gotcha.